source/loader/CMakeLists.txt

```diff
@@ -19,9 +19,9 @@ set_target_properties(loader PROPERTIES
 add_library(${PROJECT_NAME}::loader ALIAS loader)
 
 target_include_directories(loader PRIVATE
-    ${CMAKE_SOURCE_DIR}
-    ${CMAKE_SOURCE_DIR}/source/common
-    ${CMAKE_SOURCE_DIR}/source/loader
+    ${CMAKE_CURRENT_SOURCE_DIR}/../common
+    ${CMAKE_CURRENT_SOURCE_DIR}/../..
+    ${CMAKE_CURRENT_SOURCE_DIR}
```

Comment:
I actually think it would be better to use `${PROJECT_SOURCE_DIR}` which points to the root `CMakeLists.txt` in a similar way as `${CMAKE_SOURCE_DIR}` does when not part of an `add_subdirectoty`.

Reply:
This seems like the simplest solution to fixing this.